internal/server/singleprocess/service_project.go

```diff
@@ -23,6 +23,15 @@ func (s *service) UpsertProject(
 		return nil, err
 	}
 
+	if result.DataSource != nil && result.DataSource.GetGit() != nil {
+		// The project is connected to a data source so let’s
+		// try to queue some remote init operations to ensure the
+		// application list is populated and ready to work with.
+
+		// TODO: error handling
+		queueInitOps(s, ctx, result)
```

Comment:
There's a TODO here.

Also, note that this will be executed on _any_ project setting change. I think we don't have any tools right now to detect if the data source is _changed_ but we may want to add a TODO for that. Just wanted to call that out in case you didn't realize that.

Reply:
Added simple error handling to satisfy the TODO.

Yeah I’d definitely like to be more granular about when we queue this operations. Have added a TODO to capture that.